File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        longest = 0
        length = 0
        for n in s:
            if n-1 in s:
                continue
            length = 1
            while n+length in s:
                length += 1
            longest = max(longest, length)
        return max(longest, length)

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> charSet;
        int l = 0;
        int m = 0;
        for (int r = 0; r < s.size(); r++){
            if (charSet.find(s[r]) == charSet.end()){
                m = max(m, r-l+1);
            } else {
                while(charSet.count(s[r])) {
                    charSet.erase(s[l++]);
                }
            }
            charSet.insert(s[r]);
        }
        return m;
    }
};
